candidatepair.go

```diff
@@ -26,8 +26,12 @@ type candidatePair struct {
 }
 
 func (p *candidatePair) String() string {
-	return fmt.Sprintf("prio %d (local, prio %d) %s <-> %s (remote, prio %d)",
+	if p != nil {
+	  return fmt.Sprintf("prio %d (local, prio %d) %s <-> %s (remote, prio %d)",
 		p.Priority(), p.local.Priority(), p.local, p.remote, p.remote.Priority())
+	}else{
+	  return "nil"
```

Comment:
woot, excited to see you burn through these :) 

* Would you mind adding a test? Just call `String()` on a `var nilCandidatePair *candidatePair` in a test.
* I would just return ""

Reply:
Cool! I edited to return "" if the interface is nil. I'll add in the call to the function to test it out!

Reply:
Oh whoops that's not an interface:D

Reply:
Did I add a test correctly?